ilist.h triggers new -Wunused-but-set-variable variant in clang

```
In file included from ../../third_party/vulkan-deps/spirv-tools/src/tools/reduce/reduce.cpp:21:
In file included from ../../third_party/vulkan-deps/spirv-tools/src/source/opt/build_module.h:21:
In file included from ../../third_party/vulkan-deps/spirv-tools/src/source/opt/ir_context.h:30:
In file included from ../../third_party/vulkan-deps/spirv-tools/src/source/opt/cfg.h:24:
In file included from ../../third_party/vulkan-deps/spirv-tools/src/source/opt/basic_block.h:30:
In file included from ../../third_party/vulkan-deps/spirv-tools/src/source/opt/instruction_list.h:28:
../../third_party/vulkan-deps/spirv-tools/src/source/util/ilist.h:341:7: error: variable 'sentinel_count' set but not used [-Werror,-Wunused-but-set-variable]
  int sentinel_count = 0;
      ^
```

Looks like https://github.com/KhronosGroup/SPIRV-Tools/blob/master/source/util/ilist.h#L341 needs a [[maybe_unused]] if that's allowed here, or a `(void)sentinel_count` below the assert here https://github.com/KhronosGroup/SPIRV-Tools/blob/master/source/util/ilist.h#L350

@ben-clayton @dneto0 @alan-baker

ref https://bugs.chromium.org/p/chromium/issues/detail?id=1309955#c53

--- a/source/util/ilist.h
+++ b/source/util/ilist.h
@@ -348,6 +348,7 @@ void IntrusiveList<NodeType>::Check(NodeType* start) {
     p = p->next_node_;
   } while (p != start);
   assert(sentinel_count == 1 && "List should have exactly 1 sentinel node.");
+  (void)sentinel_count;
 
   p = start;
   do {

